src/frontend/next/src/pages/myfeeds.tsx

```diff
@@ -0,0 +1,200 @@
+import { useState, useEffect, FormEvent } from 'react';
+import { useRouter } from 'next/router';
+import { Box, Card, Container, Grid, IconButton, Typography } from '@material-ui/core';
+import { AccountCircle, AddCircle, RssFeed } from '@material-ui/icons';
+import { makeStyles } from '@material-ui/core/styles';
+import { ValidatorForm, TextValidator } from 'react-material-ui-form-validator';
+import { isWebUri } from 'valid-url';
+import Head from 'next/head';
+
+import config from '../config';
+import ExistingFeedList from '../components/ExistingFeedList';
+import HelpPopoverButton from '../components/HelpPopoverButton';
+import CustomizedSnackBar from '../components/SnackBar';
+import { useUser } from '../components/UserProvider';
+import { User, Feed, FeedHash } from '../interfaces';
+
+const useStyles = makeStyles((theme) => ({
+  margin: {
+    margin: theme.spacing(2),
+  },
+  button: {
+    padding: '3px 0 3px 0',
+  },
+}));
+
+const MyFeeds = () => {
+  const router = useRouter();
+  const user: User = useUser();
+  const classes = useStyles();
+  const [newFeedAuthor, setNewFeedAuthor] = useState('');
+  const [newFeedUrl, setNewFeedUrl] = useState('');
+  const [submitStatus, setSubmitStatus] = useState({ message: '', isError: false });
+  const [feedHash, updateFeedHash] = useState({});
+  const [alert, setAlert] = useState(false);
+
+  const { telescopeUrl } = config;
+
+  useEffect(() => {
+    if (!user.isLoggedIn) router.push(`${telescopeUrl}/auth/login`);
+    setNewFeedAuthor(user.name!);
+    ValidatorForm.addValidationRule('isUrl', (value: string) => !!isWebUri(value));
+    return function () {
+      ValidatorForm.removeValidationRule.bind('isUrl');
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [telescopeUrl, user.name]);
```

Comment:
This seems to depend on `user.isLoggedIn` as well, should you just use `user` here?

Reply:
@humphd : If we set to user, the `useEffect` could keep running because there would be a different user object even though the content is same. I am thinking to include just `isLoggedIn` and ignore `name`. Some similar issue with putting object as useEffect dependencies were discuss [here](https://stackoverflow.com/questions/55808749/use-object-in-useeffect-2nd-param-without-having-to-stringify-it-to-json)